core/store/src/trie/mod.rs

```diff
@@ -613,14 +614,34 @@ impl Trie {
         Ok(())
     }
 
-    // Prints the trie nodes starting from hash, up to max_depth depth.
-    // The node hash can be any node in the trie.
-    pub fn print_recursive(&self, f: &mut dyn std::io::Write, hash: &CryptoHash, max_depth: u32) {
+    /// Prints the trie nodes starting from `hash`, up to `max_depth` depth. The node hash can be any node in the trie.
+    /// Depending on arguments provided, can limit output to no more than `limit` entries,
+    /// show only subtree for a given `record_type`, or skip subtrees where `AccountId` is less than `from` or greater than `to`.
+    pub fn print_recursive(
+        &self,
+        f: &mut dyn std::io::Write,
+        hash: &CryptoHash,
+        max_depth: u32,
+        mut limit: Option<u32>,
+        record_type: Option<u8>,
+        from: Option<&AccountId>,
+        to: Option<&AccountId>,
+    ) {
         match self.debug_retrieve_raw_node_or_value(hash) {
             Ok(NodeOrValue::Node(_)) => {
                 let mut prefix: Vec<u8> = Vec::new();
-                self.print_recursive_internal(f, hash, max_depth, &mut "".to_string(), &mut prefix)
-                    .expect("write failed");
+                self.print_recursive_internal(
+                    f,
+                    hash,
+                    &mut "".to_string(),
+                    &mut prefix,
+                    max_depth,
+                    limit.as_mut(),
```

Comment:
nit: I would recommend swapping limit for a u32 here. This way you only need to deal with Option here in a single place. I hope I'm not giving you self contradicting feedback :) The rule of thumb is that the public API like print_recursive should have the type matching the user API but as soon as we reach implementation (print_recursive_internal) we can relax it a bit.